### PR TITLE
Move services to the sidebar

### PR DESCRIPTION
**Pull request by strycore**
This gets rid of the "import games" dialog and all features associated with syncing service libraries with the local one.
Now all GOG / Humble / Steam games show up in their respective section on the sidebar. With icons matching the service used (This ensures we get an icon for every game).

There has been some major refactoring in how the view is displayed, removing some coupling between the view and the database.
Since I wasn't able to get rid of Gtk critical warnings about sorting the IconView with pixbufs in the store, I delegated the sorting and filtering to the database, which makes things much easier to handle. I had a look at the music player Lollipop and it looks like all sorting is done through the database as well.

There have been some updates to the UI file and it is now possible to use Glade to edit it.

This is not 100% complete. XDG and Steam games still have to be matched correctly. But it is complete enough to go under testing.

All database stuff has been moved to a database package instead of incorrectly referring to db stuff as "PGA".

The Humble Bundle section has a tendency to segfault. It may be related to the icon format used (transparent PNG)

### CODE REVIEW

File: lutris/gui/lutriswindow.py [hunk 1]
```diff
@@ -758,40 +717,17 @@ def on_game_started(self, game):
 
     def on_game_updated(self, game):
         """Callback to refresh the view when a game is updated"""
-        logger.debug("Updating game %s", game)
+        logger.debug("%s has been updated, refreshing view", game)
         if not game.is_installed:
-            game = Game(game_id=game.id)
-            self.game_selection_changed(None, None)
+            game = Game(game_id=game.id)  # ???? Why does it need a reload?
```
Comment (AlexanderRavenheart) [inline lutris/gui/lutriswindow.py:722]: Add a TODO: as part of the comment so that this can be reviewed later.

File: lutris/gui/lutriswindow.py [hunk 2]
```diff
@@ -805,27 +741,25 @@ def game_selection_changed(self, _widget, game):
...
-        for pga_game in pga.get_games_where(slug=slug):
-            self.game_store.update(pga_game)
+        self.swap_game_panel()
 
     @GtkTemplate.Callback
     def on_add_game_button_clicked(self, *_args):
         """Add a new game manually with the AddGameDialog."""
-        self.add_popover.hide()
-        AddGameDialog(self, runner=self.selected_runner)
+        if "runner" in self.filters:
```
Comment (AlexanderRavenheart) [inline lutris/gui/lutriswindow.py:756]: This if else can be changed with:
```
       runner = self.filters.get("runner", None).
```

File: lutris/migrations/migrate_hidden_ids.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    # Turn the strings into integers
+    return [int(game_id) for game_id in ignores]
+
+
+def migrate():
+    """Run migration"""
+    try:
+        game_ids = get_hidden_ids()
+    except:
+        print("Failed to read hidden game IDs")
```
Comment (AlexanderRavenheart) [inline lutris/migrations/migrate_hidden_ids.py:21]: Why print and not logger.info or logger.warning?

File: lutris/migrations/migrate_hidden_ids.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        return []
```
Comment (AlexanderRavenheart) [inline lutris/migrations/migrate_hidden_ids.py:22]: Why return empty list? Did you mean to write: ```game_ids = []``` ?

File: lutris/runners/libretro.py [hunk 5]
```diff
@@ -197,7 +197,7 @@ def prelaunch(self):
             except (ValueError, TypeError):
                 firmware_count = 0
             system_path = self.get_system_directory(retro_config)
-            notes = retro_config["notes"] or ""
+            notes = str(retro_config["notes"] or "")
```
Comment (AlexanderRavenheart) [inline lutris/runners/libretro.py:200]: Can retro_config["notes"] be anything other than string?

File: lutris/runners/pico8.py [hunk 6]
```diff
@@ -249,10 +248,10 @@ def play(self):
...
             command.append(game_data.get("name") + " - PICO-8")
 
-            icon = datapath.get_icon_path(game_data.get("slug"))
-            if icon:
-                command.append("--icon")
-                command.append(icon)
+            # icon = datapath.get_icon_path(game_data.get("slug"))
+            # if icon:
+            #     command.append("--icon")
+            #     command.append(icon)
```
Comment (AlexanderRavenheart) [inline lutris/runners/pico8.py:254]: Why are these lines disabled?

File: lutris/services/gog.py [hunk 7]
```diff
@@ -267,34 +281,13 @@ def get_banner(cls, gog_game):
...
-    gog_service = GogService()
-    if not gog_service.is_available():
+    gog_service = GOGService()
+    if not gog_service.is_connected():
         logger.info("You are not connected to GOG")
-        connect()
-    if not gog_service.is_available():
+        gog_service.login()
+    if not gog_service.is_connected():
         raise UnavailableGame
```
Comment (AlexanderRavenheart) [inline lutris/services/gog.py:291]: No custom message for raised exception?

**Comment by tannisroot** [comment]
Doesn't launch for me
```
./bin/lutris -d
Unable to load locale dir, translations won't work.
/home/alex/git_projects/lutris/lutris/gui/application.py:27: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gio, GLib, Gtk
/home/alex/git_projects/lutris/lutris/util/display.py:13: PyGIWarning: GnomeDesktop was imported without specifying a version first. Use gi.require_version('GnomeDesktop', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gdk, GLib, GnomeDesktop
2020-08-19 23:08:05,818: MAME XML generation launched in the background, not returning anything this time
2020-08-19 23:08:05,818: Getting full game list from MAME...
2020-08-19 23:08:05,823: MAME isn't installed, can't retrieve systems list.
2020-08-19 23:08:05,895: Migrating games field hidden
INFO     2020-08-19 23:08:05,941 [application.do_command_line:299]:Running Lutris 0.5.7.1
INFO     2020-08-19 23:08:05,941 [__init__.migrate:34]:Running migration: migrate_hidden_ids
DEBUG    2020-08-19 23:08:05,944 [game.save:268]:Saving Legend of Zelda: Windwaker (dolphin) with config ID legend-of-zelda-windwaker-gamecube-1572864399
DEBUG    2020-08-19 23:08:05,949 [game.save:268]:Saving Minecraft Launcher (linux) with config ID minecraft-launcher-desktopapp
Traceback (most recent call last):
  File "/home/alex/git_projects/lutris/lutris/gui/application.py", line 300, in do_command_line
    migrate()
  File "/home/alex/git_projects/lutris/lutris/migrations/__init__.py", line 36, in migrate
    migration.migrate()
  File "/home/alex/git_projects/lutris/lutris/migrations/migrate_hidden_ids.py", line 25, in migrate
    game.hide()
  File "/home/alex/git_projects/lutris/lutris/game.py", line 132, in hide
    self.save()
  File "/home/alex/git_projects/lutris/lutris/game.py", line 268, in save
    logger.debug("Saving %s with config ID %s", self, self.config.game_config_id)
AttributeError: 'NoneType' object has no attribute 'game_config_id'
INFO     2020-08-19 23:08:05,953 [application.do_shutdown:562]:Shutting down Lutris
INFO     2020-08-19 23:08:05,953 [mame.notify_mame_xml:31]:MAME XML written
```

File: lutris/services/gog.py [hunk 8]
```diff
@@ -265,17 +282,16 @@ def new_from_gog_game(cls, gog_game):
...
-    def get_banner(cls, gog_game):
+    def get_banner(cls, gog_game, size="medium"):
         """Return the path to the game banner.
         Downloads the banner if not present.
         """
-        # Are there other formats?
-        image_url = "https:%s_prof_game_100x60.jpg" % gog_game["image"]
-        image_hash = gog_game["image"].split("/")[-1]
-        cache_dir = os.path.join(settings.CACHE_DIR, "gog/banners/small/")
+        image_url = "https:%s%s" % (gog_game["image"], GOG_BANNER_FORMATS[size])
```
Comment (AlexanderRavenheart) [inline lutris/services/gog.py:289]: Shouldn't there be // after https:, like https://%s%s?

Comment (strycore) [reply lutris/services/gog.py:289]: no it's included by GOG

File: lutris/exceptions.py [hunk 9]
```diff
@@ -31,6 +31,20 @@ def __init__(self, libraries):
...
 
+class AuthenticationError(Exception):
+    """Raised when authentication to a service fails"""
+
+
+class UnavailableGame(Exception):
+    """Raised when a game is available from a service"""
+
+
+class MultipleInstallerError(BaseException):
```
[the remark this answers is not in the document]
Comment (strycore) [reply lutris/exceptions.py:44]: it's the same thing

File: lutris/migrations/migrate_hidden_ids.py (hunk 3 above)
Comment (strycore) [reply lutris/migrations/migrate_hidden_ids.py:21]: the print statements are not here to stay

File: lutris/migrations/migrate_hidden_ids.py (hunk 4 above)
Comment (strycore) [reply lutris/migrations/migrate_hidden_ids.py:22]: no I want to exit the script

File: lutris/runners/pico8.py (hunk 6 above)
Comment (strycore) [reply lutris/runners/pico8.py:254]: the icon is missing

**Comment by strycore** [comment]
This is usable. There's likely some more fixes to be implemented but as it is, I have not found any problems installing a large number of GOG and Humble Bundle games.

I will merge this to avoid drifting away from the master branch for too long. I'm starting to receive pull requests that are incompatible with our new approach.

This is ready for user testing

**Pull request merged**